internal/config/config.go

```diff
@@ -15,22 +17,193 @@ import (
 
 type Configuration struct {
 	defaultConfig *viper.Viper
-	contextMap    map[string]*GrafanaConfig
+	AppConfig     *AppConfig
+}
+
+type AppGlobals struct {
+	Debug           bool `mapstructure:"debug" yaml:"debug"`
+	IgnoreSSLErrors bool `mapstructure:"ignore_ssl_errors" yaml:"ignore_ssl_errors"`
+}
+
+type AppConfig struct {
+	ContextName   string                       `mapstructure:"context_name" yaml:"context_name"`
+	StorageEngine map[string]map[string]string `mapstructure:"storage_engine" yaml:"storage_engine"`
+	Contexts      map[string]*GrafanaConfig    `mapstructure:"contexts" yaml:"contexts"`
+	Global        *AppGlobals                  `mapstructure:"global" yaml:"global"`
+}
+
+func (s *Configuration) ClearContexts() {
+	newContext := make(map[string]*GrafanaConfig)
+	newContext["example"] = &GrafanaConfig{
+		APIToken: "dummy",
+	}
+	appCfg := s.GetAppConfig()
+	appCfg.Contexts = newContext
+	appCfg.ContextName = "example"
+	err := s.SaveToDisk(false)
+	if err != nil {
+		log.Fatal("Failed to make save changes")
+	}
+
+	log.Info("All contexts were cleared")
+
+}
+
+// GetDefaultGrafanaConfig returns the default aka. selected grafana config
+func (s *Configuration) GetDefaultGrafanaConfig() *GrafanaConfig {
+	name := s.GetAppConfig().GetContext()
+	val, ok := s.GetAppConfig().GetContexts()[name]
+	if ok {
+		return val
+	} else {
+		log.Fatal("Context is not found.  Please check your config")
+	}
+	return nil
+
+}
+
+// CopyContext Makes a copy of the specified context and write to disk
+func (s *Configuration) CopyContext(src, dest string) {
+	//Validate context
+	contexts := s.GetAppConfig().GetContexts()
+	if len(contexts) == 0 {
+		log.Fatal("Cannot set context.  No valid configuration found in importer.yml")
+	}
+	cfg, ok := contexts[src]
+	if !ok {
+		log.Fatal("Cannot set context.  No valid configuration found in importer.yml")
```

Comment:
Same as above, more info in the error message would be helpful as to which part specifically is failing.

Reply:
Done